packages/teleport-project-generator-stencil/src/index.ts

```diff
@@ -0,0 +1,70 @@
+import { createProjectGenerator } from '@teleporthq/teleport-project-generator'
+import { createStencilComponentGenerator } from '@teleporthq/teleport-component-generator-stencil'
+import { createComponentGenerator } from '@teleporthq/teleport-component-generator'
+import { FILE_TYPE } from '@teleporthq/teleport-shared/dist/cjs/constants'
+import { createPostProcessor } from '@teleporthq/teleport-postprocessor-prettier-js'
+import { createPlugin as createImportPlugin } from '@teleporthq/teleport-plugin-import-statements'
+
+import stencilAppRouting from '@teleporthq/teleport-plugin-stencil-app-routing'
+import prettierHTML from '@teleporthq/teleport-postprocessor-prettier-html'
+
+import { Mapping } from '@teleporthq/teleport-types'
+
+import stencilProjectMapping from './stencil-mapping.json'
+
+export const createStencilProjectGenerator = () => {
+  const prettierJS = createPostProcessor({ fileType: FILE_TYPE.TSX })
+  const importStatementsPlugin = createImportPlugin({ fileType: FILE_TYPE.TSX })
+
+  const stencilComponentGenerator = createStencilComponentGenerator()
+  stencilComponentGenerator.addMapping(stencilProjectMapping as Mapping)
+
+  const routingComponentGenerator = createComponentGenerator()
+  routingComponentGenerator.addPlugin(stencilAppRouting)
+  routingComponentGenerator.addPlugin(importStatementsPlugin)
+  routingComponentGenerator.addPostProcessor(prettierJS)
+
+  const htmlFileGenerator = createComponentGenerator()
+  htmlFileGenerator.addPostProcessor(prettierHTML)
+
+  const generator = createProjectGenerator({
+    components: {
+      generator: stencilComponentGenerator,
+      path: ['src', 'components'],
+      options: {
+        createFolderForEachComponent: true,
+      },
+    },
+    pages: {
+      generator: stencilComponentGenerator,
+      path: ['src', 'components'],
+      options: {
+        createFolderForEachComponent: true,
+      },
+    },
+    router: {
+      generator: routingComponentGenerator,
+      path: ['src', 'components'],
+      fileName: 'app-root',
+    },
+    entry: {
+      generator: htmlFileGenerator,
+      appRootOverride: `<app-root></app-root>`,
+      path: ['src'],
+      fileName: 'index',
+      customScriptTags: [
+        { type: 'module', path: ['build', 'app.esm.js'] },
```

Comment:
I would keep path: string[] only when we refer to folders, not also files. I see two options:
1. path is ['build'] and there's another field fileName: 'app.esm.js'
2. path is a simple string: (build/app.esm.js). maybe this one makes more sense since the path is always the same when you know the build strategy of the project

Reply:
Makes sense, made changes 👍